### PR TITLE
[INLONG-7075][Sort] Add table level metric and dirty data backup for PostgreSQL

### LINKED ISSUE #7075: [Improve][Sort] Add table level metric and dirty data backup for PostgreSQL
### Description

Add table level metric and dirty data backup for PostgreSQL

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kuansix**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- [INLONG-7075][Sort] Add table level metric and dirty data backup for PostgreSQL
- Fixes #7075 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Comment by gong** [comment]
lost dirty metric data init

**Comment by kuansix** [comment]
> lost dirty metric data init

done. and add dirty data backup.

File: inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcMultiBatchingOutputFormat.java [hunk 1]
```diff
@@ -519,6 +512,9 @@ protected void attemptFlush() throws IOException {
                     if (!flushFlag && null != tableException) {
                         LOG.info("Put tableIdentifier:{} exception:{}",
                                 tableIdentifier, tableException.getMessage());
+                        outputMetrics(tableIdentifier, Long.valueOf(tableIdRecordList.size()),
+                                1L, true);
+                        dirtySinkHelper.invoke(record, DirtyType.RETRY_LOAD_ERROR, tableException);
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcMultiBatchingOutputFormat.java:517]: Dirty data sink should be after restart with exception when the 'isIgnoreTableException' is 'false'

File: inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcMultiBatchingOutputFormat.java [hunk 2]
```diff
@@ -519,6 +512,9 @@ protected void attemptFlush() throws IOException {
...
                         tableExceptionMap.put(tableIdentifier, tableException);
                         if (isIgnoreTableException) {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcMultiBatchingOutputFormat.java:519]: It is better to rethrow the exception when 'isIgnoreTableException' is 'false'?

**Comment by gong** [comment]
dirtySinkHelper should call open method yo init config. table is dynamic, so you can't invoke `invoke(T dirtyData, DirtyType dirtyType, Throwable e)`, maybe you need call like `setLabels("", dirtyOptions.getLabels()))`

File: inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcMultiBatchingOutputFormat.java (hunk 2 above)
Comment (kuansix) [reply inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcMultiBatchingOutputFormat.java]: has log maxRetries times before

**Comment by kuansix** [comment]
> setLabels

Will be unified processed by @Yizhou-Yang

**Review by yunqingmoswu — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Pull request merged**